cli/module_reload.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	apppb "go.viam.com/api/app/v1"
+
+	rdkConfig "go.viam.com/rdk/config"
+	"go.viam.com/rdk/logging"
+)
+
+// ModuleMap is a type alias to indicate where a map represents a module config.
+// We don't convert to rdkConfig.Module because it can get out of date with what's in the db.
+// Using maps directly also saves a lot of high-maintenance ser/des work.
+type ModuleMap map[string]interface{}
```

Comment:
Aside: go should really add a way to pass through other unknown json fields into a catch-all `map[string]any` when unmarshalling... It seems like a pretty big limitation of the stdlib. If go had that feature, I would say that we should have a type. Without it :woman_shrugging: